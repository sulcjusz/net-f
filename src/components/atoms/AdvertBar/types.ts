export interface IAdvertBarProps {
  children: React.ReactNode
  className?: string
}
