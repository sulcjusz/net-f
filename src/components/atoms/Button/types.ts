export interface IButtonProps {
  onClick?: () => void
  size?: 'small' | 'medium'
  disabled?: boolean
  children: React.ReactNode
  className?: string
}
