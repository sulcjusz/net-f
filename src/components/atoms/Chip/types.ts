export interface IChipProps {
  size?: 'small' | 'medium'
  textColor?: 'black' | 'white'
  children: React.ReactNode
  className?: string
}
