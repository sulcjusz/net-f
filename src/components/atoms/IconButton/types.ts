export interface IIconButtonProps {
  onClick?: () => void
  size?: 'small' | 'medium'
  disabled?: boolean
  icon: React.ReactNode
  className?: string
}
