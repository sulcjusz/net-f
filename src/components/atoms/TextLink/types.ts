export interface ITextLinkProps {
  text: string
  onClick: () => void
  className?: string
}
