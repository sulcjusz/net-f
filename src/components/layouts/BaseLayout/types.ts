export interface IBaseLayoutProps {
  children: React.ReactNode
}
