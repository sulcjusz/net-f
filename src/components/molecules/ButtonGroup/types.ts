export interface IButtonGroupProps {
  children: React.ReactNode
}
