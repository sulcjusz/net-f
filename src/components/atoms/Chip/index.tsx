import React from 'react'
import classnames from 'classnames'
import {IChipProps} from './types'

const Chip: React.FC<IChipProps> = ({
  size = 'medium',
  textColor = 'black',
  className,
  children
}) => {
  const buttonClasses = classnames(
    'rounded-xl border border-grey-dark font-exo leading-5 tracking-[.07em] text-center flex items-center justify-center',
    {
      'text-base py-2 px-3': size === 'small',
      'text-xl py-3 px-4 font-exo text-xs font-medium leading-6  text-center':
        size === 'medium',
      'text-grey-600': textColor === 'black',
      'text-whitw': textColor === 'white'
    },
    className
  )

  return <div className={buttonClasses}>{children}</div>
}

export default Chip
