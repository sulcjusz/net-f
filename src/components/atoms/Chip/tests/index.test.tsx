import {render, RenderResult} from '@testing-library/react'
import '@testing-library/jest-dom'

import Chip from '../index'
import {IChipProps} from '../types'

type ChipRenderProps = Partial<IChipProps>

const renderChip = (
  props?: ChipRenderProps
): {chip: HTMLElement} & RenderResult => {
  const utils = render(<Chip {...props}>Example Chip</Chip>)
  const chip = utils.getByText('Example Chip')
  return {...utils, chip}
}

describe('Chip component', () => {
  it('should match the snapshot', () => {
    const {chip} = renderChip()
    expect(chip.firstChild).toMatchSnapshot()
  })

  it('renders correctly with default props', () => {
    const {chip} = renderChip()
    expect(chip).toBeInTheDocument()
    expect(chip).toHaveClass(
      'rounded-xl border border-grey-dark font-exo text-[12px] leading-4 tracking-[.07em] text-center flex items-center justify-center'
    )
    expect(chip).toHaveClass('w-[163px] h-[53.53px] font-medium')
    expect(chip).toHaveClass('text-grey-600')
  })

  it('renders correctly with size "small"', () => {
    const {chip} = renderChip({size: 'small'})
    expect(chip).toHaveClass('w-[133.07px] h-[44.1px]')
  })

  it('renders correctly with textColor "white"', () => {
    const {chip} = renderChip({textColor: 'white'})
    expect(chip).toHaveClass('text-white')
  })
})
