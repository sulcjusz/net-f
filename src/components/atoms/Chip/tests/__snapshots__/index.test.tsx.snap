// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip component should match the snapshot 1`] = `Example Chip`;
